### PR TITLE
feat: support for multiple sdks and abstract machine making into create_image script in sdk

### PR DESCRIPTION
**Pull request by stskeeps**
This probably requires upping CARTESI_DEFAULT_SDK_VERSION as well.

### CODE REVIEW

File: apps/cli/src/commands/build.ts [hunk 1]
```diff
@@ -19,6 +19,7 @@ type ImageInfo = {
     env: string[];
     ramSize: string;
     sdkVersion: string;
+    sdkType: string;
```
Comment (endersonmaia) [inline apps/cli/src/commands/build.ts:22]: I suggest making this variable consistent with the label.

-  `sdkName`
- `CARTESI_LABEL_SDK_NAME`
- `io.cartesi.rollups.sdk_name`

File: packages/sdk/Dockerfile [hunk 2]
```diff
@@ -105,6 +105,7 @@ COPY devnet /usr/local/bin
 COPY eth_isready /usr/local/bin
 COPY eth_dump /usr/local/bin
 COPY eth_load /usr/local/bin
+COPY create_image /usr/local/bin
```
Comment (endersonmaia) [inline packages/sdk/Dockerfile:108]: `create_image` can confuse with the ext2 image and others, at least for me.

Maybe `create_snapshot` or even `create_machine_snapshot`.

File: packages/sdk/create_image [hunk 3]
```diff
@@ -0,0 +1,81 @@
+#!/bin/bash
```
Comment (endersonmaia) [inline packages/sdk/create_image:1]:
```suggestion
#!/usr/bin/env bash

set -e
```

File: packages/sdk/create_image [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+while true ; do
+    case "$1" in
+        -r|--ram-length)
+            ram_length="$2"
+            shift 2 ;;
+        -d|--drive-label)
+            drive_label="$2"
+            shift 2 ;;
+        -m|--machine-rootfs)
+            input_filename="$2"
```
Comment (endersonmaia) [inline packages/sdk/create_image:40]: I'd match variable and arguments here.

```suggestion
        -m|--drive-filename)
            drive_filename="$2"
```

**Review by endersonmaia — CHANGES_REQUESTED** [review]
Install [changesets](https://github.com/changesets/changesets) and add a changset file where you propose the minor version bump for the cartesi/sdk package.

**Review by endersonmaia — CHANGES_REQUESTED** [review]
@stskeeps changeset will take care of updating the CHANGELOG and package.json files when changes hit the main branch, a GH Actions job will create/update the Release PR.

Since this PR has changes for two packages with a dependency between them, we'd need to have the changes in separate PRs.

The first one is with changes to the cartesi/sdk package, so we can release it first. Another PR with changes to the cartesi/cli making use of the new cartesi/sdk.

**Comment by stskeeps** [comment]
OK, so close this pull request and I've opened as a start https://github.com/cartesi/cli/pull/26 - this is the style desired right?

**Pull request closed without merging**